File: app/src/main/java/com/aeonflux/app/core/media/PlaybackService.java
package com.aeonflux.app.core.media;

import android.content.Intent;
import androidx.annotation.Nullable;
import androidx.media3.exoplayer.ExoPlayer;
import androidx.media3.session.MediaLibraryService;
import androidx.media3.session.MediaSession;
import dagger.hilt.android.AndroidEntryPoint;

@AndroidEntryPoint
public class PlaybackService extends MediaLibraryService {
    private MediaLibrarySession mediaSession;
    private ExoPlayer player;

    @Override
    public void onCreate() {
        super.onCreate();
        player = new ExoPlayer.Builder(this).build();
        
        MediaLibrarySession.Callback callback = new MediaLibrarySession.Callback() {
            // Implement callbacks for media library navigation if needed
        };
        
        mediaSession = new MediaLibrarySession.Builder(this, player, callback)
            .build();
    }

    @Nullable
    @Override
    public MediaLibrarySession onGetSession(MediaSession.ControllerInfo controllerInfo) {
        return mediaSession;
    }

    @Override
    public void onDestroy() {
        if (mediaSession != null) {
            player.release();
            mediaSession.release();
            mediaSession = null;
        }
        super.onDestroy();
    }
}

File: app/src/main/java/com/aeonflux/app/core/media/PlaybackService.kt
package com.aeonflux.app.core.media

import android.content.Intent
import androidx.annotation.OptIn
import androidx.media3.common.util.UnstableApi
import androidx.media3.exoplayer.ExoPlayer
import androidx.media3.session.MediaLibraryService
import androidx.media3.session.MediaSession
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class PlaybackService : MediaLibraryService() {
    private var mediaSession: MediaLibrarySession? = null
    private lateinit var player: ExoPlayer

    @OptIn(UnstableApi::class)
    override fun onCreate() {
        super.onCreate()
        player = ExoPlayer.Builder(this).build()
        
        val callback = object : MediaLibrarySession.Callback {
            // Implement callbacks for media library navigation if needed
        }
        
        mediaSession = MediaLibrarySession.Builder(this, player, callback)
            .build()
    }

    override fun onGetSession(controllerInfo: MediaSession.ControllerInfo): MediaLibrarySession? {
        return mediaSession
    }

    override fun onDestroy() {
        mediaSession?.run {
            player.release()
            release()
            mediaSession = null
        }
        super.onDestroy()
    }
}
